### PR TITLE
Group Project : Space Travelers' Hub

### PR DESCRIPTION
**Pull request by haliljon**
## In this PR we were able to:

- Use React documentation
- Use React components
- Use React props
- Use React Router
- Connect React and Redux
- Handle events in a React app
- Write unit tests with React Testing Library
- Use styles in a React app
- Use React hooks
- Apply React best practices and language style guides in code
- Use store, actions and reducers in React
- Perform a code review for a team member

in order to create an app that works with real live data from the SpaceX API to display rockets, dragons and missions. It lets the user to reserve rockets, book dragons and join mission and allows a user to view all reserved rockets, dragons, and space missions.

## Here is the screenshot of the live page:

Screenshot 2023-02-09 at 1 14 23 AM

### CODE REVIEW

File: src/components/myProfile.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import { useSelector, useDispatch } from 'react-redux';
+import { joinMission } from '../redux/missions/missions';
+import { booking } from '../redux/dragons.js/dragons';
+import { reservation } from '../redux/rockets/rockets';
+
+const MyProfile = () => {
```
Comment (Felix45) [inline src/components/myProfile.js:6]: ## :heavy_check_mark: Great Job :clap: 
- [ ] Great job adding the following additional requirements to your project
  
- You enhanced the My Profile section by adding a placeholder message when the `My Missions`, `My dragons` or `My Rockets` lists are empty (no missions joined or no rockets reserved).
 
 - You enhanced `My Profile` section by adding the `Cancel reservation`, `Cancel dragon` and `Leave Mission` buttons to the lists here. Clicking them should dispatch the actions you have already used in the main Rockets and Missions sections.

File: src/tests/MyProfile.test.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import store from '../redux/configureStore';
+
+it('matches snapshot', () => {
+  const tree = renderer.create(
+    <Provider store={store}>
+      <MyProfile />
+    </Provider>,
+  ).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (Felix45) [inline src/tests/MyProfile.test.js:14]: - [ Required ] Great job adding `snap shot` tests to your project, however, I noticed the tests are failing when I run them. Please make sure that all your tests are passing, please see the screen shot below for more details on this and rectify.

Screenshot from 2023-02-09 00-01-22

File: src/redux/rockets/rockets.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+const ALL_ROCKETS = 'space travelers/redux/rockets/ALL_ROCKETS';
+const RESERVE_ROCKET = 'space travelers/redux/rockets/RESERVE_ROCKET';
+
+export const rocketsReducer = (state = { status: 'default', rockets: [] }, action) => {
+  switch (action.type) {
+    case `${ALL_ROCKETS}/fulfilled`:
+      // console.log(action.payload);
```
Comment (Felix45) [inline src/redux/rockets/rockets.js]: - [ Required ] Please consider removing any piece of commented code from your codebase, commented code makes your project look unprofessional. Kindly rectify this.

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @haliljon & @c00p75 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights :heavy_check_mark: 
- [x] Application is working as expected :white_check_mark: 
- [x] No linter errors :white_check_mark: 
- [x] Gitflow is used correctly :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark: 
- [x] Descriptive pull request title and summary :white_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/MyProfile.test.js (hunk 2 above)
Comment (haliljon) [reply src/tests/MyProfile.test.js:14]: Hi @Felix45, we just fixed it


Screenshot 2023-02-09 at 12 23 28 PM

File: src/redux/rockets/rockets.js (hunk 3 above)
Comment (haliljon) [reply src/redux/rockets/rockets.js]: Hi @Felix45 

Thank you for noticing that, we forgot to remove it.

**Review by prolajumokeoni — APPROVED** [review]
Hi @Team

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
-  All tests are passing✅  
- Removed  commented code 💯  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by haliljon** [comment]
Thanks @Felix45  and @prolajumokeoni  for reviewing our project 🤝🏼

**Pull request merged**